builtin/logical/consul/path_roles.go

```diff
@@ -97,34 +103,45 @@ func pathRolesRead(ctx context.Context, req *logical.Request, d *framework.Field
 	if result.Policy != "" {
 		resp.Data["policy"] = base64.StdEncoding.EncodeToString([]byte(result.Policy))
 	}
+	if len(result.Policies) > 0 {
+		resp.Data["policies"] = result.Policies
+	}
 	return resp, nil
 }
 
 func pathRolesWrite(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	tokenType := d.Get("token_type").(string)
+	policy := d.Get("policy").(string)
+	policies := d.Get("policies").([]string)
+	if len(policies) == 0 {
+		switch tokenType {
+		case "client":
+		case "management":
+		default:
+			return logical.ErrorResponse(
+				"token_type must be \"client\" or \"management\""), nil
+		}
+	}
 
-	switch tokenType {
-	case "client":
-	case "management":
-	default:
+	if (policy == "" && tokenType != "management") && len(policies) == 0 {
```

Comment:
This can move up into the if statement above. https://github.com/hashicorp/vault/pull/5586/files#diff-317d337c64fc1e40a12d6b605816d15fR116